### DISCUSSION TITLE
What am I missing here ? System stuck at "Monitor Health : System Initializing."

### DISCUSSION THREAD

**Discussion by Primer-Merc** [jgyates/genmon#1213] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

Generator Monitor Version : V1.19.04
Platform Stats :
        CPU Utilization : 2.28%
        OS Name : Debian GNU/Linux
        OS Version : 12 (bookworm
Generac EVO 2 Controller  Generac 26kw
Data Transport Method : 
         USR-TCP232-306    Serial over TCP

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have submitted logs

### What is/are your question(s)?

Not sure what's happening here. After several hours of switching wires, changing settings, I was finally to get both send/and receive from the  USR-TCP232-306  to my Genmon server instance.  As you can see from the screenshots They are connected and sending data to each other.  

****** IMPORTANT*****
I have no way of verifying if the USR-TCP232-306 is receiving/sending data from/to the EVO Controller though.  Any ideas based on the included screenshots and included logs.

 Most relevant info in the logs will be the most recent time wise. I have not changed anything for the past 2 hours ( log time). 

![Monitor Screen 3](https://github.com/user-attachments/assets/4add05ac-0129-41b0-b07e-786d52013ec2)

### Additional information

Below are screen shots of the USR web login, genmon web login, USR-MO settings page, USR-TCP test page( not receiving data) and a TeraTerm terminal screen logged into the USR. 

![Monitor Screen 1](https://github.com/user-attachments/assets/90635e76-1a3d-4551-8d2a-d887ee2842d4)
![Monitor Screen 2](https://github.com/user-attachments/assets/4affc55b-3a02-4cbe-80cc-60968aa8280f)
![Monitor Screen 3](https://github.com/user-attachments/assets/4add05ac-0129-41b0-b07e-786d52013ec2)

**Comment by Primer-Merc** [comment]
Some additional progress / information. New logs have been uploaded.  I am now getting data using the USR-TCP Test suite. 

Does the EF9D have any relationship to the modbus slave address of 9d?

![image](https://github.com/user-attachments/assets/146c5acd-9671-49e6-b85f-0f8560b217de)

**Comment by Primer-Merc** [comment, marked as answer]
Another update and I will close this.  

I decided to try and run this in a DOCKER container after watching this video. https://www.youtube.com/watch?v=w0zPj9yBtZE&t=1s

I already had CasaOS, and all the files are ready to go as an APP inside CasaOS . That fixed it. I just needed to plug in my settings for the USR TCP to Serial device in the settings menu, and everything popped up.

-Kyle-

**Reply by jgyates** [reply]
@Primer-Merc 

I am glad you got it working. Let me know if you have any other questions.
